test/integration/query-interface.test.js

```diff
@@ -455,30 +473,81 @@ describe(Support.getTestDialectTeaser('QueryInterface'), () => {
       });
     });
 
-    it('should get a list of foreign keys for the table', async function () {
+    afterEach(async function () {
+      await this.queryInterface.dropTable('hosts');
+      await this.queryInterface.dropTable('users');
+    });
 
-      const foreignKeys = await this.sequelize.query(
-        this.queryInterface.queryGenerator.getForeignKeysQuery(
-          'hosts',
-          dialect === 'db2' ? this.sequelize.config.username.toUpperCase() : this.sequelize.config.database,
-        ),
-        { type: this.sequelize.QueryTypes.FOREIGNKEYS },
-      );
+    it('should be able to retrieve list of foreign key names using getForeignKeysForTables', async function () {
+
+      const foreignKeys = await this.queryInterface.getForeignKeysForTables(['hosts'], this.options);
+      expect(foreignKeys.hosts).to.have.length(3);
+
+      switch (dialect) {
+        case 'postgres':
+          expect(foreignKeys.hosts).to.include('hosts_admin_fkey');
+          break;
+        case 'mysql':
+          expect(foreignKeys.hosts).to.include('hosts_ibfk_1');
+          break;
+        case 'sqlite':
+          expect(foreignKeys.hosts).to.include('0_0');
+          break;
+      }
+    });
 
+    // test against first and second foreign key response for data-consistency
+    it('should get a list of foreign keys for the table', async function () {
+      let SQL = this.queryInterface.queryGenerator.getForeignKeysQuery('hosts', null, this.options);
+      let foreignKeys = await this.sequelize.query(SQL, { type: this.sequelize.QueryTypes.FOREIGNKEYS });
       expect(foreignKeys).to.have.length(3);
 
-      if (dialect === 'postgres') {
-        expect(Object.keys(foreignKeys[0])).to.have.length(6);
-        expect(Object.keys(foreignKeys[1])).to.have.length(7);
-        expect(Object.keys(foreignKeys[2])).to.have.length(7);
-      } else if (['sqlite', 'db2'].includes(dialect)) {
-        expect(Object.keys(foreignKeys[0])).to.have.length(8);
-      } else if (dialect === 'ibmi') {
-        expect(Object.keys(foreignKeys[0])).to.have.length(9);
-      } else if (['mysql', 'mariadb', 'mssql'].includes(dialect)) {
-        expect(Object.keys(foreignKeys[0])).to.have.length(12);
-      } else {
-        throw new Error(`This test doesn't support ${dialect}`);
+      let firstForeignKeyProperties = Object.keys(foreignKeys[0]);
```

Comment:
use `const` as much as possible, avoid re-assigning the same variable when they serve different purposes